### PR TITLE
Loading screen permissions

### PR DESCRIPTION
**Pull request by Kimblebee**
Updated UI for Camera Permission Request
Created a reusable template that can be used when requesting other permissions in the future, i.e. microphone

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -16,7 +16,8 @@
   -->
 <resources>
     <string name="app_name">Jetpack Camera</string>
-    <string name="camera_permission_required_rationale">The camera is important for this app. Please grant the permission.</string>
-    <string name="camera_not_available">Camera not available</string>
-    <string name="request_permission">Request permission</string>
+    <string name="camera_permission_screen_title">Enable Camera</string>
+    <string name="camera_permission_required_rationale">Please provide permission to access to the camera. It is necessary for this app to function.</string>
+    <string name="image_accessibility_text">A symbol representing a camera</string>
```
Comment (yasith) [inline app/src/main/res/values/strings.xml:21]: nit: the name should be specific to the camera icon image.

eg. `camera_permission_image_accessibility_text`

File: app/src/main/java/com/google/jetpackcamera/ui/PermissionsUi.kt [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+                        .align(Alignment.Center)
+                ) {
+                    PermissionButton(
+                        //modifier = Modifier.align(Alignment.Start),
+                        permissionState = permissionState,
+                        requestButtonText = requestButtonText
+                    )
+                    Spacer(modifier = Modifier.height(20.dp))
+                    if (onSkipPermission != null) {
+                        //todo: skip permission button for optional permissions
```
Comment (yasith) [inline app/src/main/java/com/google/jetpackcamera/ui/PermissionsUi.kt:127]: //TODO(name/bug): details..

File: app/src/main/java/com/google/jetpackcamera/ui/PermissionsUi.kt [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    iconAccessibilityText: String,
+    title: String,
+    bodyText: String,
+    requestButtonText: String
+) {
+    Column(
+        modifier = modifier.background(MaterialTheme.colorScheme.primary),
+        verticalArrangement = Arrangement.Bottom
+    ) {
+        // permission image / top half
```
Comment (yasith) [inline app/src/main/java/com/google/jetpackcamera/ui/PermissionsUi.kt:83]: I think the comments you have are perfect places to extract out into sub Composables.

**Review by yasith — CHANGES_REQUESTED** [review]

**Review by yasith — COMMENTED** [review]
Can you add screenshots of the change to the PR, makes it easier to review visual changes.

**Comment by Kimblebee** [comment]
image

image
screens with/without skip button text

**Comment by yasith** [comment]
> image
> 
> image screens with/without skip button text

This looks very nice!!

File: app/src/main/java/com/google/jetpackcamera/ui/PermissionsUi.kt [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+                    .align(Alignment.CenterHorizontally)
+                    .height(IntrinsicSize.Min)
+            ) {
+                // permission buttons
+                Column(
+                    modifier = Modifier
+                        .align(Alignment.Center)
+                ) {
+                    PermissionButton(
+                        //modifier = Modifier.align(Alignment.Start),
```
[the remark this answers is not in the document]
Comment (Kimblebee) [reply app/src/main/java/com/google/jetpackcamera/ui/PermissionsUi.kt]: yeah just leftovers when i was working through the layout. removed

File: app/src/main/java/com/google/jetpackcamera/ui/PermissionsUi.kt (hunk 2 above)
Comment (Kimblebee) [reply app/src/main/java/com/google/jetpackcamera/ui/PermissionsUi.kt]: actually no longer necessary

File: app/src/main/java/com/google/jetpackcamera/ui/PermissionsUi.kt [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+                requestButtonText = requestButtonText,
+                onSkipPermission = onSkipPermission
+            )
+        }
+        Spacer(modifier = Modifier.fillMaxHeight(.2f))
+    }
+}
+
+@Composable
+
```
Comment (yasith) [inline app/src/main/java/com/google/jetpackcamera/ui/PermissionsUi.kt:117]: nit: extra new line

**Review by yasith — APPROVED** [review]

**Pull request merged**